anchore_engine/services/policy_engine/engine/feeds/download.py

```diff
@@ -127,10 +123,61 @@ def read(self, feed, group, start_index):
         ]
         return FileListJsonIterator(files, start_index)
 
+    def _get_group_metadata(self, feed, group):
+        for results in self.metadata.download_result.results:
+            if results.feed == feed and results.group == group:
+                return results.group_metadata
+
+    def read_files(self, feed, group):
+        files = [
+            os.path.join(self.root_dir, feed, group, x)
+            for x in os.listdir(os.path.join(self.root_dir, feed, group))
+        ]
```

Comment:
yep! the LocalFeedDataRepo appears to store the files for a given feed/group combination in a specific directory. So the intent here is to provide access to the files in that specific directory.

Reply:
And so far as i can tell, that directory has a flat structure. The existing read() method in LocalFeedDataRepo treats it as such.

Reply:
we're calling os.path.join to generate the path to the specific directory in question. then we're calling os.listdir on that directory. the directory is flat, meaning it should only have files in it. So i don't think this is an issue.

Reply:
Ah I see what you mean. You aren't expecting that directory to ever have directories in it, only files. In that case, the helper is assuming that is the case (flat directory). If I was using `FileListIterator` somewhere else, I could miss that it will return anything including directories which is not intended - causing breakage because it will try to read the directories. 
There are a couple of ways to make this explicit about files. One is to use `os.path.isfile()` or use `os.walk()` which separates the files from the directories.

Reply:
While i'm not expecting anyone else to use `FileListIterator`, this is good input. I will opt for the `os.path.isfile()` check inside of FileListIterator. If i used `os.walk()`, it would be in `LocalFeedDataRepo.read_files`, which would not support downstream consumers of `FileListIterator`

Reply:
updated

Reply:
This is great. I understand that the current use doesn't directly benefit from this change, but this is great to _prevent_ future problems and allow anyone else to use this and take advantage of the iterator without worrying about directories. Thank you for taking the time to put this change in